java/code/src/com/suse/manager/ssl/SSLCertManager.java

```diff
@@ -117,4 +121,36 @@ public SSLCertPair generateCertificate(SSLCertPair caPair, String password, SSLC
             }
         }
     }
+
+    /**
+     * Extract the CN and alternate DNS names from an SSL certificate.
+     *
+     * @param certificate the certificate to read
+     * @return the set of subject names
+     */
+    public Set<String> getNamesFromSslCert(String certificate) {
+        String output = execHelper.exec(List.of("openssl", "x509", "-noout", "-subject", "-ext", "subjectAltName"),
+                certificate);
```

Comment:
We are not using openssl directly anywhere else, only indirectly. Shouldn't we, just to be safe, also add openssl dependency on the spacewalk-java?

Also, isn't there a java native tooling for certificates?

Reply:
hum... right we probably need it. In fact it looked silly to me to call salt runner to call openssl for this. For native libs [X509Certificate](https://docs.oracle.com/javase/8/docs/api/javax/security/cert/X509Certificate.html) seems not able to extract alternate names and that means adding BouncyCastle as a dependency... I'ld prefer direct `openssl` then.